### PR TITLE
KAZOO-3499 alternative lookup by account realm

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/fax/src/fax_smtp.erl [hunk 1]
```diff
@@ -458,6 +459,82 @@ maybe_faxbox_owner(#state{faxbox=FaxBoxDoc}=State) ->
             end
     end.
 
+-spec maybe_faxbox_domain(state()) -> state().
+maybe_faxbox_domain(#state{faxbox_email=Domain}=State) ->
+    ViewOptions = [{'key', Domain}],
+    case couch_mgr:get_results(?WH_ACCOUNTS_DB, <<"accounts/listing_by_realm">>, ViewOptions) of
+        {'ok', [JObj]} -> AccountId = wh_json:get_value([<<"value">>, <<"account_id">>], JObj),
+                          maybe_faxbox_by_owner_email(AccountId, State);
+        _ ->
```
Comment (jamesaimonetti) [inline applications/fax/src/fax_smtp.erl]: Is there any value in differentiating `{'ok', []}` vs `{'error', _E}` in the case clauses?

File: applications/crossbar/priv/couchdb/account/users.json [hunk 2]
```diff
@@ -16,7 +16,10 @@
        },
        "creds_by_sha": {
            "map": "function(doc) { if(!doc.pvt_sha1_auth || (doc.pvt_type != 'user' || doc.pvt_deleted)) return; emit( doc.pvt_sha1_auth, {'owner_id': doc._id, 'account_id': doc.pvt_account_id} ); }"
-       }
+       },
+       "list_by_email": {
+            "map": "function(doc) { if( doc.pvt_type != 'user' || doc.pvt_deleted) return; if( !doc.email || doc.email == '' ) return ;emit( doc.email, {'owner_id': doc._id, 'account_id': doc.pvt_account_id} ); }"
```
Comment (jamesaimonetti) [inline applications/crossbar/priv/couchdb/account/users.json]: Why not just `emit(doc.email, null);` ? `doc._id` will be in the `id` of the view result, and this is in the account's DB, right? So we already know the account.

Comment (lazedo) [inline applications/crossbar/priv/couchdb/account/users.json]: followed the logic for other views like creds_by_sha, do you want to change all the views ?

File: applications/fax/src/fax_smtp.erl (hunk 1 above)
Comment (lazedo) [inline applications/fax/src/fax_smtp.erl]: i don't think there is a value here.

File: applications/crossbar/priv/couchdb/account/users.json (hunk 2 above)
Comment (jamesaimonetti) [inline applications/crossbar/priv/couchdb/account/users.json]: No, just this one would be fine by me, for now.

File: applications/fax/src/fax_smtp.erl [hunk 3]
```diff
@@ -458,6 +459,82 @@ maybe_faxbox_owner(#state{faxbox=FaxBoxDoc}=State) ->
...
+-spec maybe_faxbox_by_owner_email(ne_binary(), state()) -> state().
+maybe_faxbox_by_owner_email(AccountId, #state{from=From}=State) ->
+    ViewOptions = [{'key', From}],
+    AccountDb = wh_util:format_account_db(AccountId),
+    case couch_mgr:get_results(AccountDb, <<"users/list_by_email">>, ViewOptions) of
+        {'ok', []} -> 
+            lager:debug("user ~s does not exist in account ~s, trying by rules",[From, AccountId]),
+            maybe_faxbox_by_rules(AccountId, State);
+        {'ok', [JObj]} -> 
+            OwnerId = wh_json:get_value([<<"value">>, <<"owner_id">>], JObj),
```
Comment (jamesaimonetti) [inline applications/fax/src/fax_smtp.erl]: If the view changes, this becomes `wh_json:get_value(<<"id">>, JObj)`

**Pull request merged**